App crashed due to "Failed resolution of: Landroid/icu/number/NumberFormatter$UnitWidth"

### Steps to reproduce

1. Open Breezy Weather normally
2. Wait until app loaded

### Expected behavior

The app should open properly & shows up the current weather forecast. 

### Actual behavior

App immediately crashed. Crash log can be see on the "Other details" section. 

### Weather source used

BMKG (because my current location suggested this weather source) 

### Breezy Weather version

6.0.1-ALPHA

### Android version

Android 7.0

### Device

Xiaomi Redmi Note 4 (mido) 

### Other details

Here is the crash log when the app crashed. I discovered this from F-Droid, and when I updated to this Alpha version, it crashed immediately. Maybe it was a bug on the temperature data component or something? 

`java.lang.NoClassDefFoundError: Failed resolution of: Landroid/icu/number/NumberFormatter$UnitWidth;
	at A0.j.h(SourceFile:1)
	at org.breezyweather.common.basic.models.options.basic.UnitUtils.formatMeasure(SourceFile:110)
	at org.breezyweather.common.basic.models.options.basic.UnitUtils.formatMeasure$default(SourceFile:24)
	at org.breezyweather.common.basic.models.options.unit.TemperatureUnit.formatMeasure(SourceFile:3)
	at org.breezyweather.common.basic.models.options.appearance.DetailDisplay.getCurrentValue(SourceFile:299)
	at androidx.navigation.compose.q.k(SourceFile:38)
	at x6.e.r(SourceFile:296)
	at org.breezyweather.ui.main.fragments.HomeFragment.O(SourceFile:281)
	at org.breezyweather.ui.main.fragments.j.emit(SourceFile:243)
	at kotlinx.coroutines.flow.f0.a(SourceFile:198)
	at org.breezyweather.ui.main.fragments.k$a.invokeSuspend(SourceFile:43)
	at org.breezyweather.ui.main.fragments.k$a.invoke(SourceFile:2)
	at org.breezyweather.ui.main.fragments.k$a.invoke(SourceFile:1)
	at androidx.lifecycle.RepeatOnLifecycleKt$repeatOnLifecycle$3$1$1$1$1$1$1.invokeSuspend(SourceFile:33)
	at androidx.lifecycle.RepeatOnLifecycleKt$repeatOnLifecycle$3$1$1$1$1$1$1.invoke(SourceFile:2)
	at androidx.lifecycle.RepeatOnLifecycleKt$repeatOnLifecycle$3$1$1$1$1$1$1.invoke(SourceFile:1)
	at androidx.work.O.f0(SourceFile:19)
	at kotlinx.coroutines.D.g(SourceFile:11)
	at androidx.lifecycle.RepeatOnLifecycleKt$repeatOnLifecycle$3$1$1$1$1.invokeSuspend(SourceFile:78)
	at n3.a.resumeWith(SourceFile:9)
	at N3.b.h(SourceFile:132)
	at a.a.Q(SourceFile:15)
	at kotlinx.coroutines.B.invoke(SourceFile:129)
	at kotlinx.coroutines.D.t(SourceFile:35)
	at androidx.lifecycle.RepeatOnLifecycleKt$repeatOnLifecycle$3$1$1$1.onStateChanged(SourceFile:30)
	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(SourceFile:25)
	at androidx.lifecycle.LifecycleRegistry.forwardPass(SourceFile:91)
	at androidx.lifecycle.LifecycleRegistry.sync(SourceFile:72)
	at androidx.lifecycle.LifecycleRegistry.moveToState(SourceFile:33)
	at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(SourceFile:15)
	at M0.b0.p(SourceFile:41)
	at M0.b0.k(SourceFile:56)
	at M0.T.L(SourceFile:62)
	at M0.T.u(SourceFile:36)
	at M0.B.onStart(SourceFile:60)
	at g.k.onStart(SourceFile:1)
	at org.breezyweather.ui.main.MainActivity.onStart(SourceFile:1)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1249)
	at android.app.Activity.performStart(Activity.java:6871)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2710)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2808)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1541)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:165)
	at android.app.ActivityThread.main(ActivityThread.java:6375)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:912)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:802)
	Suppressed: N3.f: [w0{Cancelling}@2fa7103, Dispatchers.Main.immediate]
Caused by: java.lang.ClassNotFoundException: Didn't find class "android.icu.number.NumberFormatter$UnitWidth" on path: DexPathList[[zip file "/data/app/org.breezyweather-2/base.apk"],nativeLibraryDirectories=[/data/app/org.breezyweather-2/lib/arm64, /data/app/org.breezyweather-2/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:74)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:380)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	... 49 more
`

### Acknowledgements

- [x] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue.
- [x] I checked that my issue is not covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document.
- [x] I have updated the app to **[version 5.4.8 or later](https://github.com/breezy-weather/breezy-weather/releases)** and I can reproduce the issue in this version.
- [x] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language. I can write in my native language under the English text, if necessary.
- [x] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

--- a/app/src/main/java/org/breezyweather/common/basic/models/options/basic/UnitUtils.kt
+++ b/app/src/main/java/org/breezyweather/common/basic/models/options/basic/UnitUtils.kt
@@ -169,17 +169,18 @@ object UnitUtils {
             // LogHelper.log(msg = "Formatting with ICU ${enum.id}: ${enum.measureUnit} per ${enum.perMeasureUnit}")
 
             // If result is null, it skips to the default non-ICU formatter
-            getAdjustedFormatting(
+            val adjustedFormatting = getAdjustedFormatting(
                 context.currentLocale,
                 enum.measureUnit!!,
                 unitWidth
-            )?.let { (locale, numberFormatterWidth, measureFormatWidth) ->
+            )
+            if (adjustedFormatting != null) {
                 if (enum.measureUnit is TimeUnit && unitWidth != UnitWidth.NARROW) {
                     return formatDurationWithIcu(
                         context,
                         convertedValue,
                         enum.measureUnit as TimeUnit,
-                        measureFormatWidth!!
+                        adjustedFormatting.third!!
                     )
                 }
 
@@ -187,21 +188,21 @@ object UnitUtils {
                     SettingsManager.getInstance(context).useNumberFormatter
                 ) {
                     formatWithNumberFormatter(
-                        locale,
+                        adjustedFormatting.first,
                         convertedValue,
                         enum.measureUnit!!,
                         enum.perMeasureUnit,
                         precision,
-                        numberFormatterWidth!!
+                        adjustedFormatting.second!!
                     )
                 } else {
                     formatWithMeasureFormat(
-                        locale,
+                        adjustedFormatting.first,
                         convertedValue,
                         enum.measureUnit!!,
                         enum.perMeasureUnit,
                         precision,
-                        measureFormatWidth!!
+                        adjustedFormatting.third!!
                     )
                 }
             }

